idf_component_manager/core.py

```diff
@@ -168,8 +171,8 @@ def create_project_from_example(self, example, path=None):
         print_info('Example "{}" successfully downloaded to {}'.format(example_name, os.path.abspath(project_path)))
 
     @general_error_handler
-    def add_dependency(self, dependency_str, component='main'):  # type: (str, str) -> None
-        manifest_filepath, _ = self._get_manifest(component)
+    def add_dependency(self, dependency_str, component='main', component_path=None):  # type: (str, str, str) -> None
```

Comment:
Even though we added the new CLI - `compote`, project-related functions are available through `idf.py` too and the same set of options must be supported in `idf_component_manager/idf_extensions.py` too. 

The format of idf.py extensions is briefly described here. https://github.com/espressif/esp-idf/blob/master/tools/idf_py_actions/README.md

Reply:
Still working on this one...